src/utils/getSequenceFlatMap.ts

```diff
@@ -1,6 +1,68 @@
-import { OrderObject } from '../parser/types';
+import { ComponentBlock } from '../parser/types';
 import { Sequence } from '../store/types';
 
-export function getSequenceFlatMap<T extends Sequence | OrderObject>(sequence: T): string[] {
+export function getSequenceFlatMap<T extends Sequence | ComponentBlock>(sequence: T): string[] {
   return sequence.components.flatMap((component) => (typeof component === 'string' ? component : getSequenceFlatMap(component)));
 }
+
+type SkipConditionWithExtents = {
+  currentBlock: Sequence;
+  firstIndex: number;
+  lastIndex: number;
+};
+
+function _findBlockForStep(sequence: Sequence, step: number, distance: number): (SkipConditionWithExtents[]) | number {
+  let componentsSeen = 0;
+  for (let i = 0; i < sequence.components.length; i += 1) {
+    const component = sequence.components[i];
+    if (typeof component === 'string') {
+      if (step === distance + componentsSeen) {
```

Comment:
Not sure how you feel about this -- I always do this because it makes it more readable for me. Up to you.
```suggestion
      if (step === (distance + componentsSeen) ) {
```